.eslintrc

```diff
@@ -0,0 +1,21 @@
+{
+  "overrides": [
+    {
+      "files": "*.ts",
+      "parser": "@typescript-eslint/parser",
+      "rules": {
+        "no-unused-vars": "off",
+        "no-useless-constructor": "off",
+        "@typescript-eslint/no-unused-vars": "error",
```

Comment:
I'd like to move towards using [gts](https://www.npmjs.com/package/gts) rather than a custom `.eslintrc`, but this will be a breaking change.

@mleguen realistically, do you think we'll be able to convert `yargs-parser` and `yargs/yargs.js` without calling it breaking?

I'm wondering if we should release a new version of yargs, with what's been converted so far, and plan on this final work being a major bump.

Reply:
There are quite a few style differences when using `gts`. Such as:
- semicolons
- trailing commas
- bracket spacing
- arrow body style
- keyword spacing

There would be over 3,000 problems that would need to be fixed. Most are automatically fixable with the `--fix` option. However this seems like the type of big-bang PR you are wanting to avoid. So you might consider a more gradual transition such as only applying it in the `lib` folder or on a per rule type of change like adding semicolons. Then once all the code has been updated to `gts` doing a cleanup of the lint config to just have `gts` related config.

Reply:
I recommend:
- switching to ts first, keeping eslint at the moment
- then drop nodejs 8 support and switch to gts for linting, in a carefully reviewed big-bang PR. I already did a "dry-run" of this on yargs a few months ago, and it is not that difficult to review. IMHO doing it gradually would bring more cons than pros.

Reply:
> realistically, do you think we'll be able to convert yargs-parser and yargs/yargs.js without calling it breaking?

I'd like too. But I will know for sure if it is possible when I will start working on yargs' index.d.ts